### PR TITLE
Multi attribute semantic search. Add partial scores and unify formatting of attributes and names.

### PR DESCRIPTION
**Pull request by AndreyPikunov**
**Partial scores for explainability**
Major addition is partial scores for superlinked results.
There are two places where I've added them.
I'm highlighting them in comments to this PR. 

**Formatting**
Attributes are formatted as `look`, `habitat`, and `behavior`.
Monsters are formatted as simple as Whispering Shade (no ").

### CODE REVIEW

File: docs/articles/multi-attribute-semantic-search.md [hunk 1]
```diff
@@ -316,12 +316,20 @@ app.query(
...
 | Luminoth         | 0.340084 | Moth-like creature with glowing wings and antenna              | Dense forests and jungles with bioluminescent flora | Emits soothing light patterns to communicate and attract prey |
 | Living Graffiti  | 0.330587 | Two-dimensional, colorful creature that inhabits flat surfaces | Urban areas, particularly walls and billboards      | Shapeshifts to blend with surroundings and absorbs pigments   |
 
-Et voila! This time, each of our top returned monsters' ranks highly in a score that represents a kind of "mean" of all three of the characteristics we want our monster to have. Our second and third results possess all three characteristics, and our top result (Whispering Shade), though it's `behavior` is less related to `light manipulation`, is _very_ relevant in `look` (`glowing`) and `habitat` (`dark places`) characteristics, giving it the highest score overall. What an improvement!
+Et voila! This time, each of our top returned monsters ranks highly in a score that represents a kind of "mean" of all three characteristics we want our monster to have. Here's a detailed breakdown of the scores:
```
Comment (AndreyPikunov) [inline docs/articles/multi-attribute-semantic-search.md:324]: ➡️
Here is a new paragraph with partial scores and their explanation.

File: docs/articles/multi-attribute-semantic-search.md [hunk 2]
```diff
@@ -418,12 +426,27 @@ weights = {
...
 | Zephyr Dancer  | 0.136456 | Graceful avian creature with iridescent feathers                     | High mountain peaks and wind-swept plains           | Creates mesmerizing aerial displays to attract mates           |
 
-Great results again! Our two other retrieved monsters - Luminoth and Zephyr Dancer - have behavior similar to Harmonic Coral, and live in habitats different from Harmonic Coral's. They also look very different from Harmonic Coral. (Harmonic's tendrils and Luminoth's antenna are similar features, but we only down-weighted `look_weight` by 0.5, and the resemblance between the two monsters ends there.)
+Great results again! Our two other retrieved monsters—Luminoth and Zephyr Dancer—have behavior similar to Harmonic Coral and live in habitats different from Harmonic Coral's. They also look very different from Harmonic Coral. (While Harmonic Coral's tendrils and Luminoth's antenna are somewhat similar features, we only down-weighted the `look_weight` by 0.5, and the resemblance between the two monsters ends there.)
```
Comment (AndreyPikunov) [inline docs/articles/multi-attribute-semantic-search.md:434]: ➡️
Here is a new paragraph with partial scores and their explanation.

**Review by morkapronczay — APPROVED** [review]

File: docs/articles/multi-attribute-semantic-search.md [hunk 3]
```diff
@@ -418,12 +426,27 @@ weights = {
...
+|                |     look |   habitat | behavior |    total |
+| :------------- | -------: | --------: | -------: | -------: |
+| Harmonic Coral |  0.19245 |   -0.3849 |   0.3849 |  0.19245 |
+| Luminoth       | 0.052457 | -0.068144 | 0.164884 | 0.149196 |
+| Zephyr Dancer  | 0.050741 | -0.079734 | 0.165449 | 0.136456 |
+
+The negative score in the `habitat` attribute indicates a down-weighting, reflecting our intention to de-emphasize `habitat` similarity in this particular query.
+
+Both Luminoth and Zephyr Dancer maintain strong `behavior` scores comparable to Harmonic Coral, indicating similar behavioral traits. Their `look` scores are positive but lower, reflecting some visual similarities while still maintaining distinct appearances. The negative `habitat` scores emphasize their different living environments compared to Harmonic Coral.
+
```
Comment (robertdhayanturner) [inline docs/articles/multi-attribute-semantic-search.md:447]: @andrey
negative scores in `habitat` result from down-weighting, per your statement above. Seems like this is a magnitude comparison (not a negative vs positive comparison).

Comment (AndreyPikunov) [reply docs/articles/multi-attribute-semantic-search.md]: You are right, we should say that we pushing similar monsters **away** and not just down-weight. Probably then like this?

The negative score for the `habitat` attribute means that monsters with similar habitat are not just down-weighted but rather "pushed away" from the Harmonic Coral during the search.

**Pull request merged**